src/input/KeyEncoder.zig

```diff
@@ -413,6 +414,19 @@ fn legacy(
         return try std.fmt.bufPrint(buf, "\x1B{c}", .{byte});
     }
 
+    // If we are on macOS, command+keys do not encode text. It isn't
+    // typical for command+keys on macOS to ever encode text. They
+    // don't in native text inputs (i.e. TextEdit) and they also don't
+    // in other native terminals (Terminal.app officially but also
+    // iTerm2).
+    //
+    // For Linux, we continue to encode text because it is typical.
+    // For example on Gnome Console Super+b will encode a "b" character
+    // with legacy encoding.
+    if ((comptime builtin.os.tag == .macos) and all_mods.super) {
+        return "";
+    }
```

Comment:
I will note that it is possible that this check is too late. We may want to omit ctrl sequence and other encodings as well when the super key is pressed on macOS. But the tests only cover this and this was the least-impacting change I could do with a very specific real world bug to verify it resolves. 

I'd be open to a broader change with a bit more research.